ijServer/src/main/kotlin/org/jetbrains/research/tasktracker/Utils.kt

```diff
@@ -21,3 +33,34 @@ fun initializeIds() {
     currentResearchIdAtomic = AtomicInteger(ResearchTable.maxId() + 1)
     currentSurveyIdAtomic = AtomicInteger(0)
 }
+
+suspend inline fun PipelineContext<Unit, ApplicationCall>.createLogFile(
+    subDirectory: String,
+    crossinline insert: (name: String, researchId: Int) -> Unit
+) {
+    val multipartData = call.receiveMultipart()
+    val researchIndex = call.parameters.getOrFail<Int>("id")
+    multipartData.forEachPart { part ->
+        when (part) {
+            is PartData.FileItem -> {
+                val fileName = part.originalFileName as String
+                val fileBytes = part.streamProvider().readBytes()
+                insert(fileName, researchIndex)
+                getFile(researchIndex, fileName, subDirectory).writeBytes(fileBytes)
+            }
+
+            else -> {}
+        }
+        part.dispose()
+    }
+    call.respond(HttpStatusCode.Accepted)
+}
+
+fun getFile(researchId: Int, filename: String, subDirectory: String): File {
+    val userId = transaction { ResearchTable.select { ResearchTable.id.eq(researchId) }.first()[ResearchTable.userId] }
+    val directoryPath = "files/$userId/$researchId/$subDirectory"
```

Comment:
I guess it is better to store data by researchers: "files/$researchId/$userId/$subDirectory" since you probably need to download data for a particular research, not for a user